### PR TITLE
Allow gem to install h3 as a native extension

### LINKED ISSUE #2: Add ability to automatically install h3 C
It's possible to install h3 from source as part of the gem installation. We can (and probably should) script this for convenience. 

The Python bindings already do this https://github.com/uber/h3-py/blob/master/.install.sh

### PR DESCRIPTION
**Pull request by seanhandley**
Bit hacky, but it works.

Fixes #2 🎉

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -1,3 +1,27 @@
...
+    `git submodule update --init --recursive`
+    print "Building h3..."
+    `cd ext/h3; make > /dev/null 2>&1`
+    puts " done."
+  end
+end
+
+desc "Remove compiled H3 library"
+task :clean do
+  File.delete("ext/h3/src/Makefile") if File.exists?("ext/h3/src/Makefile")
```
Comment (fxn) [inline Rakefile]: I wonder... the `Makefile` of H3 has a `clean` target. It does not delete the directories themselves, but wouldn't it be enough to delegate to their `make clean` and assume they know what it needs to be cleaned?

File: lib/h3/bindings/base.rb [hunk 2]
```diff
@@ -8,7 +8,7 @@ def self.extended(base)
         base.extend FFI::Library
         base.include Structs
         base.include Types
-        base.ffi_lib ["libh3", "libh3.1"]
+        base.ffi_lib ["ext/h3/src/lib/libh3.dylib", "ext/h3/src/lib/libh3.so"]
```
Comment (fxn) [inline lib/h3/bindings/base.rb]: Do these relative paths work when a user uses the gem?

Comment (seanhandley) [reply lib/h3/bindings/base.rb]: Yes. Tested it locally.

File: Rakefile (hunk 1 above)
Comment (seanhandley) [reply Rakefile]: Good thinking - I'll see if it works 👍

**Comment by seanhandley** [comment]
Regarding packaging - no, there's no binary packages at all for H3 and the other bindings assume it isn't preinstalled so go about downloading/building it directly.

**Comment by seanhandley** [comment]
Since we include it as a git submodule we can move HEAD to the correct place each time we release a new version.

File: Rakefile (hunk 1 above)
Comment (seanhandley) [reply Rakefile]: Tried it - it doesn't seem to remove any of the compiled code...

File: lib/h3/bindings/base.rb (hunk 2 above)
Comment (seanhandley) [reply lib/h3/bindings/base.rb]: FYI I tested this badly!

After releasing `3.3.0` I added it to `stuart-api` and discovered a `LoadError`. The paths do indeed need to be absolute 🤦‍♂️ 

Hotfixed here: https://github.com/StuartApp/h3_ruby/commit/d192b41c945ba9e1c255aa8922d8067f3f84b1f6

I yanked `3.3.0` from Rubygems and released `3.3.1`. Seems to work just fine now 😅

**Pull request merged**